### PR TITLE
53 add endpoint to delete room types

### LINKED ISSUE #53: Create rest endpoint to delete Room Type.
Create rest endpoint with method DELETE to delete room types
The controller path should be /admin/room-types/{id} where id is the id of deleted room type. It should be in admin package in controllers Look at https://github.com/dorm-social-room-manager/dsrm-backend/issues/69

Endpoint should return 204 if room type was deleted and 404 if not found.
When deleting room type we should also set all room with this type to to type null.

Test scenarios
If room type exists then it should be deleted. All dependent rooms should be without room type.
If room type does not exists it should return 404

### PR DESCRIPTION
**Pull request by Krizzuu**
closes: #53

### CODE REVIEW

File: src/main/java/com/dsrm/dsrmbackend/services/impl/RoomTypeServiceImpl.java [hunk 1]
```diff
@@ -1,34 +1,51 @@
...
+        return roomTypeRepo.findById(roomId);
     }
 
     public Page<RoomType> getRoomTypes(Pageable pageable) {
-        return repository.findAll(pageable);
+        return roomTypeRepo.findAll(pageable);
+    }
+
+    @Override
+    public Optional<RoomType> deleteRoomType(String id) {
```
Comment (HerbMat) [inline src/main/java/com/dsrm/dsrmbackend/services/impl/RoomTypeServiceImpl.java]: We want to use cascade delete like here https://www.javatpoint.com/jpa-cascade-remove

Comment (Krizzuu) [reply src/main/java/com/dsrm/dsrmbackend/services/impl/RoomTypeServiceImpl.java]: Added CascadeType.REMOVE but it seems it does not set null in Room entity but it removes it too

File: src/test/java/com/dsrm/dsrmbackend/admin/AdminRoomTypeControllerTests.java [hunk 2]
```diff
@@ -72,4 +73,27 @@ public void tryToAddNamelessRoomType() throws Exception {
...
+                .andExpect(jsonPath("$.name").value("Sala telewizyjna"));
+        this.mockMvc.perform(get("/room-types/2")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+        this.mockMvc.perform(get("/rooms/3")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.roomType").value(nullValue()));
+    }
+
```
Comment (Adrian-Rochminski) [inline src/test/java/com/dsrm/dsrmbackend/admin/AdminRoomTypeControllerTests.java]: It seems to me that checking for the occurrence of a record in the database after a deletion has been performed was supposed to be done using repo and not perform

File: src/test/java/com/dsrm/dsrmbackend/admin/AdminRoomTypeControllerTests.java [hunk 3]
```diff
@@ -72,4 +73,27 @@ public void tryToAddNamelessRoomType() throws Exception {
...
+    @Test
+    public void tryToDeleteNonExistentRoomType() throws Exception {
+        this.mockMvc.perform(delete("/admin/room-types/2222")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+    }
 }
```
Comment (Adrian-Rochminski) [inline src/test/java/com/dsrm/dsrmbackend/admin/AdminRoomTypeControllerTests.java]: as above + I think the test is redundant since we are checking a non-existent record anyway

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1,2 +1,4 @@
+spring.jpa.defer-datasource-initialization=true
+spring.sql.init.mode=always
```
Comment (Adrian-Rochminski) [inline src/main/resources/application.properties]: unnecessary

File: src/main/java/com/dsrm/dsrmbackend/controllers/admin/AdminRoomTypeController.java [hunk 5]
```diff
@@ -35,4 +40,12 @@ public ResponseEntity<Void> addRoomType(@Valid @RequestBody RoomTypeRequestDTO r
...
     }
+
+    @DeleteMapping(value="/room-types/{id}")
+    public ResponseEntity<RoomType> deleteRoomType(@PathVariable String id) {
+        Optional<RoomType> roomType = roomTypeService.deleteRoomType(id);
+        return roomType.map(type -> new ResponseEntity<>(type, HttpStatus.NO_CONTENT))
+                .orElseGet(() -> new ResponseEntity<>(null, HttpStatus.NOT_FOUND));
+
+    }
 }
```
Comment (Adrian-Rochminski) [inline src/main/java/com/dsrm/dsrmbackend/controllers/admin/AdminRoomTypeController.java]: unnecessary. Method should only return void and HttpStatus.NO_CONTENT

File: src/main/java/com/dsrm/dsrmbackend/entities/RoomType.java [hunk 6]
```diff
@@ -16,4 +18,12 @@ public class RoomType {
     @Id
     private String id;
     private String name;
+    @JsonIgnore
```
Comment (Adrian-Rochminski) [inline src/main/java/com/dsrm/dsrmbackend/entities/RoomType.java]: unnecessary

File: src/main/java/com/dsrm/dsrmbackend/services/RoomTypeService.java [hunk 7]
```diff
@@ -12,5 +12,7 @@ public interface RoomTypeService {
     Optional<RoomType> getRoomType(String roomId);
 
     Page<RoomType> getRoomTypes(Pageable pageable);
+
+    Optional<RoomType> deleteRoomType(String id);
```
Comment (Adrian-Rochminski) [inline src/main/java/com/dsrm/dsrmbackend/services/RoomTypeService.java]: it should return void

File: src/main/java/com/dsrm/dsrmbackend/services/impl/RoomTypeServiceImpl.java [hunk 8]
```diff
@@ -16,19 +16,26 @@
...
```
Comment (Adrian-Rochminski) [inline src/main/java/com/dsrm/dsrmbackend/services/impl/RoomTypeServiceImpl.java]: as above

**Review by Adrian-Rochminski — CHANGES_REQUESTED** [review]

**Review by Adrian-Rochminski — DISMISSED** [review]

File: src/main/java/com/dsrm/dsrmbackend/services/impl/RoomTypeServiceImpl.java [hunk 9]
```diff
@@ -16,19 +16,25 @@
...
     }
 
     public Page<RoomType> getRoomTypes(Pageable pageable) {
-        return repository.findAll(pageable);
+        return roomTypeRepo.findAll(pageable);
+    }
+
+    @Override
+    public void deleteRoomType(String id) {
+        Optional<RoomType> roomType = roomTypeRepo.findById(id);
```
Comment (HerbMat) [inline src/main/java/com/dsrm/dsrmbackend/services/impl/RoomTypeServiceImpl.java]: ```roomTypeRepo.deleteById```

**Review by HerbMat — CHANGES_REQUESTED** [review]

File: src/test/java/com/dsrm/dsrmbackend/RoomControllerTests.java [hunk 10]
```diff
@@ -43,7 +43,6 @@ public void getExistentRoom() throws Exception {
                 .andExpect(status().isOk())
                 .andExpect(jsonPath("$.roomNumber").value(111))
                 .andExpect(jsonPath("$.floor").value(1))
-                .andExpect(jsonPath("$.roomType.id").value(1))
```
Comment (HerbMat) [inline src/test/java/com/dsrm/dsrmbackend/RoomControllerTests.java:46]: why did you remove it?

File: src/main/java/com/dsrm/dsrmbackend/services/impl/RoomTypeServiceImpl.java [hunk 11]
```diff
@@ -16,19 +16,24 @@
...
+        return roomTypeRepo.findById(roomId);
     }
 
     public Page<RoomType> getRoomTypes(Pageable pageable) {
-        return repository.findAll(pageable);
+        return roomTypeRepo.findAll(pageable);
+    }
+
+    @Override
+    public void deleteRoomType(String id) {
```
Comment (HerbMat) [inline src/main/java/com/dsrm/dsrmbackend/services/impl/RoomTypeServiceImpl.java:38]: ```@Transactional``` missing

**Review by HerbMat — CHANGES_REQUESTED** [review]

**Review by HerbMat — APPROVED** [review]
